This file is part of MXE. See LICENSE.md for licensing information.

libccdb crashes in strlen() under windows because s = getenv("HOME"); returns NULL

diff -Naur libcddb-1.3.2.orig/lib/cddb_conn.c libcddb-1.3.2/lib/cddb_conn.c
--- libcddb-1.3.2.orig/lib/cddb_conn.c	2006-10-15 13:53:11.000000000 +0200
+++ libcddb-1.3.2/lib/cddb_conn.c	2016-12-07 19:31:32.016000000 +0100
@@ -100,6 +100,13 @@
         c->use_cache = CACHE_ON;
         /* construct cache dir '$HOME/[DEFAULT_CACHE]' */
         s = getenv("HOME");
+        if (s == NULL)
+            s = getenv("TEMP");
+        if (s == NULL)
+            s = getenv("TMP");
+        if (s == NULL)
+			s = ".";
+
         c->cache_dir = (char*)malloc(strlen(s) + 1 + sizeof(DEFAULT_CACHE) + 1);
         sprintf(c->cache_dir, "%s/%s", s, DEFAULT_CACHE);
         c->cache_read = FALSE;
